File: lightbulb-easyrecyclerview/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerview/decorations/VerticalAndHorizontalSpaceItemDecoration.java
package com.github.rooneyandshadows.lightbulb.easyrecyclerview.decorations;

import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class VerticalAndHorizontalSpaceItemDecoration extends RecyclerView.ItemDecoration {

    private final int verticalSpacing;
    private final int horizontalSpacing;

    public VerticalAndHorizontalSpaceItemDecoration(int spacing) {
        this.verticalSpacing = spacing;
        this.horizontalSpacing = spacing;
    }

    public VerticalAndHorizontalSpaceItemDecoration(int verticalSpacing, int horizontalSpacing) {
        this.verticalSpacing = verticalSpacing;
        this.horizontalSpacing = horizontalSpacing;
    }

    @Override
    public void getItemOffsets(Rect outRect, @NonNull View view, RecyclerView parent, @NonNull RecyclerView.State state) {
        outRect.left = horizontalSpacing;
        outRect.right = horizontalSpacing;
        outRect.top = verticalSpacing;
        if (parent.getChildAdapterPosition(view) == parent.getAdapter().getItemCount() - 1)
            outRect.bottom = verticalSpacing;
    }
}

File: lightbulb-easyrecyclerview/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerview/decorations/VerticalAndHorizontalSpaceItemDecoration.kt
package com.github.rooneyandshadows.lightbulb.easyrecyclerview.decorations

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView


class VerticalAndHorizontalSpaceItemDecoration(
    private val verticalSpacing: Int,
    private val horizontalSpacing: Int,
) : RecyclerView.ItemDecoration() {

    constructor(spacing: Int) : this(spacing, spacing)

    @Override
    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        outRect.left = horizontalSpacing
        outRect.right = horizontalSpacing
        outRect.top = verticalSpacing
        if (parent.getChildAdapterPosition(view) == parent.adapter!!.itemCount - 1) outRect.bottom = verticalSpacing
    }
}